grammars/css.cson

```diff
@@ -20,294 +20,1235 @@
+                  }
+                ]
+              }
+              {
+                'include': '#url'
+              }
+              {
+                'include': '#commas'
+              }
+              {
+                'include': '#comment-block'
+              }
+              {
+                'include': '#escapes'
+              }
+            ]
+          }
+          {
+            'begin': '{'
+            'beginCaptures':
+              '0':
+                'name': 'punctuation.section.document.begin.bracket.curly.css'
+            'end': '}'
+            'endCaptures':
+              '0':
+                'name': 'punctuation.section.document.end.bracket.curly.css'
+            'name': 'meta.at-rule.document.body.css'
+            'patterns': [
+              {
+                'include': '$self'
+              }
+            ]
+          }
+        ]
+      }
+      {
+        # @keyframes
+        'begin': '(?i)(?=@keyframes([\\s\'"{;]|/\\*|$))'
+        'end': '(?<=})'
+        'patterns': [
+          {
+            'begin': '(?i)\\G(@)keyframes'
+            'beginCaptures':
+              '0':
+                'name': 'keyword.control.at-rule.keyframes.css'
+              '1':
+                'name': 'punctuation.definition.keyword.css'
+            'end': '(?=\\s*{)'
+            'name': 'meta.at-rule.keyframes.header.css'
+            'patterns': [
+              {
+                'include': '#comment-block'
+              }
+              {
+                'include': '#escapes'
+              }
+              {
+                'match': '''(?x)
+                  (?:[-a-zA-Z_]    | [^\\x00-\\x7F])     # First letter
+                  (?:[-a-zA-Z0-9_] | [^\\x00-\\x7F]      # Remainder of identifier
+                    |\\\\(?:[0-9a-fA-F]{1,6}|.)
+                  )*
+                '''
+                'name': 'variable.parameter.keyframe-list.css'
+              }
+            ]
+          }
+          {
+            'begin': '{'
+            'beginCaptures':
+              '0':
+                'name': 'punctuation.section.keyframes.begin.bracket.curly.css'
+            'end': '}'
+            'endCaptures':
+              '0':
+                'name': 'punctuation.section.keyframes.end.bracket.curly.css'
+            'name': 'meta.at-rule.keyframes.body.css'
+            'patterns': [
+              {
+                'include': '#comment-block'
+              }
+              {
+                'include': '#escapes'
+              }
+              {
+                'captures':
+                  '1':
+                    'name': 'entity.other.keyframe-offset.css'
+                  '2':
+                    'name': 'entity.other.keyframe-offset.percentage.css'
+                'match': '''(?xi)
+                  (?<![\\w-]) (from|to) (?![\\w-])         # Keywords for 0% | 100%
+                  |
+                  ([-+]?(?:\\d+(?:\\.\\d+)?|\\.\\d+)%)     # Percentile value
```

Comment:
Care to make this stricter to match only percentages from 0% to 100%, or tokenize percentages smaller or higher as invalid? Up to you.